react-youtube-adaptive-loading/src/containers/Watch/WatchContent/WatchContent.js

```diff
@@ -29,41 +29,47 @@ import { getCommentsForVideo } from '../../../store/reducers/comments';
 import './WatchContent.scss';
 import { DEV_MODE } from '../../../config';
 
-class WatchContent extends Component {
-  shouldShowLoader = () => {
-    const { nextPageToken } = this.props;
+const WatchContent = ({
+  nextPageToken,
+  videoId,
+  video,
+  channel,
+  relatedVideos,
+  comments,
+  amountComments,
+  bottomReachedCallback,
+  isHeavyExperience
+}) => {
+  const shouldShowLoader = () => {
     return !!nextPageToken;
   };
 
-  render() {
-    let { videoId, video, channel, relatedVideos, comments, amountComments, bottomReachedCallback } = this.props;
-
-    if (DEV_MODE) {
-      videoId = '3C0btIJjrW8';
-      video = require('../../../dummy-data/watch/video.json');
-      channel = require('../../../dummy-data/watch/channel.json');
-      relatedVideos = require('../../../dummy-data/watch/related-videos.json');
-      comments = require('../../../dummy-data/watch/comments.json');
-      amountComments = 9473;
-    }
-
-    if (!videoId) {
-      return <div />
-    }
+  if (DEV_MODE) {
+    videoId = '3C0btIJjrW8';
+    video = require('../../../dummy-data/watch/video.json');
+    channel = require('../../../dummy-data/watch/channel.json');
+    const dummyRelatedVideos = require('../../../dummy-data/watch/related-videos.json');
+    relatedVideos = isHeavyExperience ? dummyRelatedVideos : dummyRelatedVideos.slice(0, dummyRelatedVideos.length / 2);
+    comments = isHeavyExperience ? require('../../../dummy-data/watch/comments.json') : [];
+    amountComments = 9473;
```

Comment:
What does `amountComments` correspond to? Total comments fetched? rendered?... just curious here :)

Reply:
It's just the total number of comments and it does not mean that we fetch the payload for all the comments but just the total number of comments. Just one value payload as meta data.
It's obtained by a different request from the one for fetching full comments data.